Fallback resource type without title cause panic

## Issue detail
Fallback resource type without title cause panic

## Reproduce process

### command / error messages
```
$ go run cmd/awsdac/main.go minimal-panic.yaml

WARN[0000] Type AWS::EC2::UndefinedType is not defined in the DAC definition file. It's fall backed to its service icon (Type AWS::EC2). 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x58 pc=0x102abb3ec]

goroutine 1 [running]:
github.com/awslabs/diagram-as-code/internal/types.(*Resource).SetLabel(...)
        /Users/kitayu/GitHub/diagram-as-code/branches/fix-panic/internal/types/resource.go:192
github.com/awslabs/diagram-as-code/internal/ctl.loadResources(0x14000120600, {0x140000358d8?}, 0x14000035a88)
        /Users/kitayu/GitHub/diagram-as-code/branches/fix-panic/internal/ctl/create.go:276 +0x47c
github.com/awslabs/diagram-as-code/internal/ctl.CreateDiagramFromDacFile({0x16d66f489, 0x12}, 0x14000030550, 0x14000035b58)
        /Users/kitayu/GitHub/diagram-as-code/branches/fix-panic/internal/ctl/dacfile.go:140 +0x740
main.main.func2(0x140001ae300?, {0x140000305e0?, 0x7?, 0x102b32800?})
        /Users/kitayu/GitHub/diagram-as-code/branches/fix-panic/cmd/awsdac/main.go:81 +0x130
github.com/spf13/cobra.(*Command).execute(0x140001b4008, {0x1400001e0d0, 0x1, 0x1})
        /Users/kitayu/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:983 +0x7ec
github.com/spf13/cobra.(*Command).ExecuteC(0x140001b4008)
        /Users/kitayu/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x344
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/kitayu/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
main.main()
        /Users/kitayu/GitHub/diagram-as-code/branches/fix-panic/cmd/awsdac/main.go:98 +0x348
exit status 2
Self exited with status: exit status: 1
```

### Yaml template
```
Diagram:
  DefinitionFiles:
    - Type: URL
      Url: https://raw.githubusercontent.com/awslabs/diagram-as-code/main/definitions/definition-for-aws-icons-light.yaml
  Resources:
    Canvas:
      Type: AWS::Diagram::Canvas
      Children:
        - AWSCloud
    AWSCloud:
      Type: AWS::Diagram::Cloud
      Children:
        - UndefinedType
    UndefinedType:
      Type: AWS::EC2::UndefinedType
```

--- a/cmd/awsdac-mcp-server/main.go
+++ b/cmd/awsdac-mcp-server/main.go
@@ -268,12 +268,11 @@ func handleGenerateDiagram(
 	// Create output file path
 	outputFile := filepath.Join(tempDir, "output.png")
 
-	// Generate diagram directly in the main thread
-	// This ensures logs are properly captured
+	// Generate diagram with panic recovery
 	opts := &ctl.CreateOptions{
 		OverwriteMode: ctl.Force, // Use Force for temporary files
 	}
-	if err := ctl.CreateDiagramFromDacFile(inputFile, &outputFile, opts); err != nil {
+	if err := createDiagramSafely(inputFile, &outputFile, opts); err != nil {
 		return nil, fmt.Errorf("failed to create diagram: %v", err)
 	}
 
@@ -337,11 +336,11 @@ func handleGenerateDiagramToFile(
 		return nil, fmt.Errorf("failed to write input file: %v", err)
 	}
 
-	// Generate diagram - save directly to specified path
+	// Generate diagram with panic recovery
 	opts := &ctl.CreateOptions{
 		OverwriteMode: ctl.NoOverwrite, // MCP server refuses to overwrite existing files
 	}
-	if err := ctl.CreateDiagramFromDacFile(inputFile, &outputFilePath, opts); err != nil {
+	if err := createDiagramSafely(inputFile, &outputFilePath, opts); err != nil {
 		return nil, fmt.Errorf("failed to create diagram: %v", err)
 	}
 
@@ -378,6 +377,16 @@ func handleGenerateDacFromUserRequirements(
 			},
 		},
 	}, nil
+}
+
+// createDiagramSafely wraps ctl.CreateDiagramFromDacFile with panic recovery
+func createDiagramSafely(inputFile string, outputFile *string, opts *ctl.CreateOptions) (err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("panic occurred during diagram creation: %v", r)
+		}
+	}()
+	return ctl.CreateDiagramFromDacFile(inputFile, outputFile, opts)
 }
 
 // readPromptFile reads a prompt file from the embedded filesystem

--- a/internal/ctl/create.go
+++ b/internal/ctl/create.go
@@ -270,11 +270,6 @@ func loadResources(template *TemplateStruct, ds definition.DefinitionStructure, 
 				}
 				log.Warnf("Type %s is not defined in the DAC definition file. It's fall backed to its service icon (Type %s).\n", v.Type, newType)
 				def = fallbackDef
-
-				// Change the title to indicate the original resource type for fallback icons.
-				if v.Title == "" {
-					resources[k].SetLabel(&v.Type, nil, nil)
-				}
 			}
 			if def == nil {
 				log.Warnf("Definition for %s is nil. Skip this resource.\n", v.Type)

--- a/internal/ctl/create_test.go
+++ b/internal/ctl/create_test.go
@@ -1,0 +1,182 @@
+package ctl
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/awslabs/diagram-as-code/internal/definition"
+	"github.com/awslabs/diagram-as-code/internal/types"
+)
+
+func TestLoadResourcesWithUndefinedTypeFallback(t *testing.T) {
+	// Create definition structure with fallback definition
+	ds := definition.DefinitionStructure{
+		Definitions: map[string]*definition.Definition{
+			"AWS::EC2": {
+				Type: "Resource",
+				Label: &definition.DefinitionLabel{
+					Title: "EC2",
+				},
+			},
+		},
+	}
+
+	// Create template with undefined type
+	template := &TemplateStruct{
+		Diagram: Diagram{
+			Resources: map[string]Resource{
+				"TestResource": {
+					Type: "AWS::EC2::UndefinedType", // This type doesn't exist
+				},
+			},
+		},
+	}
+
+	actualResources := make(map[string]*types.Resource)
+
+	// Define expected test resource with label set
+	expectedTestResource := new(types.Resource).Init()
+	labelText := "EC2"
+	expectedTestResource.SetLabel(&labelText, nil, nil)
+
+	// This should not panic and should use fallback
+	err := loadResources(template, ds, actualResources)
+	if err != nil {
+		t.Fatalf("loadResources failed: %v", err)
+	}
+
+	// Debug: Print full objects
+	t.Logf("actualResources: %+v", actualResources)
+	t.Logf("expectedTestResource: %+v", expectedTestResource)
+
+	// Verify TestResource was created via fallback
+	actualTestResource, exists := actualResources["TestResource"]
+	if !exists {
+		t.Fatal("TestResource was not created via fallback")
+	}
+
+	if actualTestResource == nil {
+		t.Fatal("TestResource is nil")
+	}
+
+	// Debug: Print actualTestResource details
+	t.Logf("actualTestResource: %+v", actualTestResource)
+
+	// Verify it's the same type as expected (both are *types.Resource)
+	if reflect.TypeOf(actualTestResource) != reflect.TypeOf(expectedTestResource) {
+		t.Errorf("TestResource type mismatch. Expected: %T, Got: %T", expectedTestResource, actualTestResource)
+	}
+
+	// Deep check: Compare object contents
+	isEqual := reflect.DeepEqual(actualTestResource, expectedTestResource)
+	t.Logf("DeepEqual result: %v", isEqual)
+	if !isEqual {
+		t.Errorf("TestResource deep comparison failed.\nExpected: %+v\nActual: %+v", expectedTestResource, actualTestResource)
+	}
+}
+
+func TestLoadResourcesWithDefinedType(t *testing.T) {
+	// Create definition structure with defined type
+	ds := definition.DefinitionStructure{
+		Definitions: map[string]*definition.Definition{
+			"AWS::EC2::Instance": {
+				Type: "Resource",
+				Label: &definition.DefinitionLabel{
+					Title: "EC2 Instance",
+				},
+			},
+		},
+	}
+
+	// Create template with defined type (no fallback needed)
+	template := &TemplateStruct{
+		Diagram: Diagram{
+			Resources: map[string]Resource{
+				"TestResource": {
+					Type: "AWS::EC2::Instance", // This type exists
+				},
+			},
+		},
+	}
+
+	actualResources := make(map[string]*types.Resource)
+
+	// Define expected test resource with label set
+	expectedTestResource := new(types.Resource).Init()
+	labelText := "EC2 Instance"
+	expectedTestResource.SetLabel(&labelText, nil, nil)
+
+	// This should not panic and should use direct definition
+	err := loadResources(template, ds, actualResources)
+	if err != nil {
+		t.Fatalf("loadResources failed: %v", err)
+	}
+
+	// Verify TestResource was created without fallback
+	actualTestResource, exists := actualResources["TestResource"]
+	if !exists {
+		t.Fatal("TestResource was not created")
+	}
+
+	if actualTestResource == nil {
+		t.Fatal("TestResource is nil")
+	}
+
+	// Verify it's the same type as expected
+	if reflect.TypeOf(actualTestResource) != reflect.TypeOf(expectedTestResource) {
+		t.Errorf("TestResource type mismatch. Expected: %T, Got: %T", expectedTestResource, actualTestResource)
+	}
+
+	// Deep check: Compare object contents
+	if !reflect.DeepEqual(actualTestResource, expectedTestResource) {
+		t.Errorf("TestResource deep comparison failed.\nExpected: %+v\nActual: %+v", expectedTestResource, actualTestResource)
+	}
+}
+
+func TestLoadResourcesWithNoFallbackPossible(t *testing.T) {
+	// Create definition structure without fallback definition
+	ds := definition.DefinitionStructure{
+		Definitions: map[string]*definition.Definition{
+			"AWS::S3::Bucket": {
+				Type: "Resource",
+				Label: &definition.DefinitionLabel{
+					Title: "S3 Bucket",
+				},
+			},
+		},
+	}
+
+	// Create template with undefined type that cannot fallback
+	template := &TemplateStruct{
+		Diagram: Diagram{
+			Resources: map[string]Resource{
+				"TestResource": {
+					Type: "AWS::NonExistent::Service", // Neither this nor AWS::NonExistent exists
+				},
+			},
+		},
+	}
+
+	actualResources := make(map[string]*types.Resource)
+
+	// This should not panic but should skip the resource
+	err := loadResources(template, ds, actualResources)
+	if err != nil {
+		t.Fatalf("loadResources failed: %v", err)
+	}
+
+	// Verify TestResource was NOT created (skipped due to no fallback)
+	_, exists := actualResources["TestResource"]
+	if exists {
+		t.Error("TestResource should not have been created when no fallback is possible")
+	}
+
+	// Verify only Canvas was created
+	if len(actualResources) != 1 {
+		t.Errorf("Expected 1 resource (Canvas only), got %d", len(actualResources))
+	}
+
+	if actualResources["Canvas"] == nil {
+		t.Error("Canvas resource should still be created")
+	}
+}

